controlplane/kubeadm/internal/controllers/fakes_test.go

```diff
@@ -80,65 +80,69 @@ func (f fakeWorkloadCluster) ForwardEtcdLeadership(_ context.Context, _ *cluster
 }
 
-func (f fakeWorkloadCluster) ReconcileEtcdMembers(_ context.Context, _ []string, _ semver.Version) ([]string, error) {
+func (f fakeWorkloadCluster) ReconcileEtcdMembers(context.Context, []string, semver.Version) ([]string, error) {
 	return nil, nil
 }
 
-func (f fakeWorkloadCluster) ClusterStatus(_ context.Context) (internal.ClusterStatus, error) {
+func (f fakeWorkloadCluster) ClusterStatus(context.Context) (internal.ClusterStatus, error) {
 	return f.Status, nil
 }
 
-func (f fakeWorkloadCluster) GetAPIServerCertificateExpiry(_ context.Context, _ *bootstrapv1.KubeadmConfig, _ string) (*time.Time, error) {
+func (f fakeWorkloadCluster) GetAPIServerCertificateExpiry(context.Context, *bootstrapv1.KubeadmConfig, string) (*time.Time, error) {
 	return f.APIServerCertificateExpiry, nil
 }
 
-func (f fakeWorkloadCluster) AllowBootstrapTokensToGetNodes(_ context.Context) error {
+func (f fakeWorkloadCluster) AllowBootstrapTokensToGetNodes(context.Context) error {
 	return nil
 }
 
-func (f fakeWorkloadCluster) AllowClusterAdminPermissions(_ context.Context, _ semver.Version) error {
+func (f fakeWorkloadCluster) AllowClusterAdminPermissions(context.Context, semver.Version) error {
 	return nil
 }
 
-func (f fakeWorkloadCluster) ReconcileKubeletRBACRole(_ context.Context, _ semver.Version) error {
+func (f fakeWorkloadCluster) ReconcileKubeletRBACRole(context.Context, semver.Version) error {
 	return nil
 }
 
-func (f fakeWorkloadCluster) ReconcileKubeletRBACBinding(_ context.Context, _ semver.Version) error {
+func (f fakeWorkloadCluster) ReconcileKubeletRBACBinding(context.Context, semver.Version) error {
 	return nil
 }
 
-func (f fakeWorkloadCluster) UpdateKubernetesVersionInKubeadmConfigMap(_ context.Context, _ semver.Version) error {
+func (f fakeWorkloadCluster) UpdateKubernetesVersionInKubeadmConfigMap(semver.Version) func(*bootstrapv1.ClusterConfiguration, *[]string) {
 	return nil
 }
 
-func (f fakeWorkloadCluster) UpdateEtcdVersionInKubeadmConfigMap(_ context.Context, _, _ string, _ semver.Version) error {
+func (f fakeWorkloadCluster) UpdateEtcdVersionInKubeadmConfigMap(string, string) func(*bootstrapv1.ClusterConfiguration, *[]string) {
 	return nil
 }
 
-func (f fakeWorkloadCluster) UpdateKubeletConfigMap(_ context.Context, _ semver.Version) error {
+func (f fakeWorkloadCluster) UpdateKubeletConfigMap(context.Context, semver.Version) error {
 	return nil
 }
 
-func (f fakeWorkloadCluster) RemoveEtcdMemberForMachine(_ context.Context, _ *clusterv1.Machine) error {
+func (f fakeWorkloadCluster) RemoveEtcdMemberForMachine(context.Context, *clusterv1.Machine) error {
 	return nil
 }
 
-func (f fakeWorkloadCluster) RemoveMachineFromKubeadmConfigMap(_ context.Context, _ *clusterv1.Machine, _ semver.Version) error {
+func (f fakeWorkloadCluster) RemoveMachineFromKubeadmConfigMap(context.Context, *clusterv1.Machine, semver.Version) error {
 	return nil
 }
 
-func (f fakeWorkloadCluster) EtcdMembers(_ context.Context) ([]string, error) {
```

Comment:
these changes around `_` seem like side cleanups that could have been avoided in the diff?

Reply:
Wondering how they came up. If it is via linter: our CI should have already brought this up. I guess this is whitelisted in golangci-lint config? (If so we shouldn't change it in this PR?!)

Reply:
since I was adding / modifying two functions, wanted to align to [recently merged changes](https://github.com/kubernetes-sigs/cluster-api/pull/10124/files) in the repo, otherwise two different standard would have been followed in this file.

@chrischdi both named (with `_`) and unnamed unused parameters are allowed by the linter, however it is just a good practice to not have named parameters until atleast one function parameter is in use.

Reply:
Is there a "standard" to follow from CAPI or core k8s which is defined as preferred one?

OK with the change, just makes it harder to review the PR.

Reply:
No standard AFAIK, it's just a good practice. I'm also fine with only modifying necessary functions in the test file and create a chore to follow one standard repo wide, for unused unnamed parameters - whichever is easier? 🙂

Reply:
@adityabhatia let's drop unnecessary changes, so we have a smaller changset (this PR is a good candidate for backporting, and a small changeset could make it easier)